code/modules/vending/vending.dm

```diff
@@ -1587,6 +1602,50 @@
 		product_list += new/datum/data/vending_product(/obj/item/reagent_containers/food/snacks/breakfast, rand(2, 4), hidden=1)
 		product_list += new/datum/data/vending_product(/obj/item/reagent_containers/food/snacks/snack_cake, rand(1, 3), hidden=1)
 
+/datum/data/vending_product/player_product
+	var/contents = list()
+	var/product_type
+	var/real_name
+	product_amount = 1
+	New(obj/item/product,var/price)
+		..()
+		product_type = product.type
+		product_name = product.name
+		real_name = product.real_name
+		contents += product
+		product_cost = price
+//The burden of these machinations weighs on my shoulders
+//And thus you will be burdened
+/obj/machinery/vending/player
+	name = "YouVend"
+	icon_state = "player"
```

Comment:
hey so like i know this is somewhat out of scope, but it'd be cool if we could like change the appearance a bit

idea:
not sure on the actual mechanics, but hitting the machine with an item (maybe like drag drop? not sure.) would add an overlay of the item (50% scale, would be distorted unlike this) onto the machine
![image](https://user-images.githubusercontent.com/4741640/113760724-6e6b8600-96cb-11eb-81a4-f005c17fdba7.png)

Reply:
Or what if it cycled through vendable objects?

Reply:
I think Azrun's ideal is perfect, but I think it's out of scope for the initial PR! I like to compartmentalize my changes- not to mention get merged sooner :)

Reply:
You can now choose an item in the service panel as the icon

Reply:
You can now select products in the service menu as the promo icon